app/forms/gobierto_admin/admin_form.rb

```diff
@@ -23,7 +23,8 @@ class AdminForm
 
     validates :name, :email, presence: true
     validates :email, format: { with: Admin::EMAIL_ADDRESS_REGEXP }
-    validates :password, presence: true, confirmation: true
+    validates :password, presence: true, if: Proc.new { |f| f.admin.new_record? }
+    validates :password, confirmation: true, if: Proc.new { |f|  f.password != f.admin.password }
```

Comment:
@ferblape because code is worth than thousand words, I've just pushed my thoughts here 😛 Would you say this is still readable and should work as expected?